src/byte_functions/mod.rs

```diff
@@ -271,11 +267,31 @@ fn to_hex_decimal(chr: char) -> u8 {
         '8' => 8,
         '9' => 9,
         'a' => 10,
+        'A' => 10,
         'b' => 11,
+        'B' => 11,
         'c' => 12,
+        'C' => 12,
         'd' => 13,
+        'D' => 13,
         'e' => 14,
+        'E' => 14,
         'f' => 15,
+        'F' => 15,
         _ => 0, // Fucked up but yolo for now
     }
 }
+
+fn two_char_to_byte_v2(c1: char, c2: char) -> u8 {
+    return to_hex_decimal_v2(c1) * 16 + to_hex_decimal_v2(c2);
+}
+
+fn to_hex_decimal_v2(chr: char) -> u8 {
+
+    let x = chr as u8;
+    let lower_4 = x & 0x0f;
+    let is_letter = (x & 0x40) >> 6;
+    
+    return lower_4 + is_letter * 9;
```

Comment:
We should probably add some comments / ref as to how the bitwise operations work to achieve the desired result, especially in relation to the ASCII codes and binary stuffs